src/java/org/apache/cassandra/tools/nodetool/DataPaths.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.cassandra.tools.nodetool;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import io.airlift.airline.Arguments;
+import io.airlift.airline.Command;
+import io.airlift.airline.Option;
+import org.apache.cassandra.tools.NodeProbe;
+import org.apache.cassandra.tools.NodeTool.NodeToolCmd;
+import org.apache.cassandra.tools.nodetool.stats.DataPathsHolder;
+import org.apache.cassandra.tools.nodetool.stats.DataPathsPrinter;
+import org.apache.cassandra.tools.nodetool.stats.StatsPrinter;
+
+@Command(name = "datapaths", description = "Print all directories where data of tables are stored")
+public class DataPaths extends NodeToolCmd
+{
+    @Arguments(usage = "[<keyspace.table>...]", description = "List of table (or keyspace) names")
+    private List<String> tableNames = new ArrayList<>();
+
+    @Option(title = "format", name = {"-F", "--format"}, description = "Output format (json, yaml)")
```

Comment:
@rtib why is this upper cased? "F" ? What is wrong with just "f". Is it used somewhere already?

Reply:
There is nothing wrong with the lower case "f" option, I guess. But, TableStats, TpStats and CompactionStats are using it identically. That is the whole bunch of subcommands built on the StatsHolder/StatsPrinter infrastructure of org.apache.cassandra.tools.nodetool.stats. Even if datapaths subcommand is not a statistics feature, I found the use of StatsHolder/StatsPrinter useful, convenient and a suitable pattern.

Reply:
fair enough.